### DISCUSSION TITLE
adding <> to mini.pairs

### DISCUSSION THREAD

**Discussion by jaredmcqueen** [LazyVim/LazyVim#1163] [Q&A] [answered]
I write a lot of rust and would really like letter than and greater than `<>` pair completion.

I added a file called `mini-pairs.lua` file that contains:
```lua
return {
  {
    "echasnovski/mini.nvim",
    opts = {
      mappings = {
        ["<"] = { action = "open", pair = "<>", neigh_pattern = "[^\\]." },
        [">"] = { action = "close", pair = "<>", neigh_pattern = "[^\\]." },
      },
    },
  },
}
```

this did not work.  Reading through the help shows the following:
```
- Insert `<>` pair if `<` is typed at line start, don't register for `<CR>`: >
  lua MiniPairs.map('i', '<', { action = 'open', pair = '<>', neigh_pattern = '\r.', register = { cr = false } })
  lua MiniPairs.map('i', '>', { action = 'close', pair = '<>', register = { cr = false } })
```

when I added the following two lines at the end of my `init.lua` I still got errors:
```lua
MiniPairs.map("i", "<", { action = "open", pair = "<>", neigh_pattern = "\r.", register = { cr = false } })
MiniPairs.map("i", ">", { action = "close", pair = "<>", register = { cr = false } })
```

At this point I have no idea how to configure mini.pairs and need some help, any nudge in the right direction would be greatly appreciated!

**Comment by jyuan0** [comment, marked as answer]
Have you tried with `"echasnovski/mini.pairs"` instead of `"echasnovski/mini.nvim"` (the standalone plugin instead of the library)?

**Reply by folke** [reply]
This is the correct answer. Good catch!

**Reply by jaredmcqueen** [reply]
thanks!
